Library/Homebrew/style.rb

```diff
@@ -164,17 +164,24 @@ def run_rubocop(files, output_type,
       end
     end
 
-    def run_shellcheck(files, output_type)
+    def run_shellcheck(files, output_type, fix: false)
       files = shell_scripts if files.blank?
 
-      args = ["--shell=bash", "--enable=all", "--external-sources", "--source-path=#{HOMEBREW_LIBRARY}", "--", *files]
+      files = files.map(&:realpath)
+
+      args = ["--shell=bash", "--enable=all", "--external-sources", "--source-path=#{HOMEBREW_LIBRARY}", "--"]
+
+      if fix
+        patch = system_command shellcheck, args: ["--format=diff", *args, *files]
+        system_command "patch", args: ["-d", "/", "-p0"], input: patch.stdout
+      end
```

Comment:
Is there a better way to apply patches? This system call requires a `patch` command. The patch file generated by `shellcheck` uses absolute paths which cannot apply by `git apply`.